extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -1247,22 +1257,28 @@ export class AmpDatePicker extends AMP.BaseElement {
    * @param {!JsonObject} param
    */
   onDatesChange(param) {
+    const isSameDay = this.ReactDates_['isSameDay'];
     const startDate = param['startDate'];
     const endDate = param['endDate'];
     const isFinalSelection = (!this.props_['keepOpenOnDateSelect'] &&
         this.state_['focusedInput'] != this.ReactDatesConstants_['END_DATE']);
 
-    let containsBlocked = false;
-
+    let blockedCount = 0;
     if (startDate && !this.allowBlockedRanges_) {
       this.iterateDateRange_(startDate, endDate, index => {
         if (this.blocked_.contains(index)) {
-          containsBlocked = true;
+          blockedCount++;
         }
       });
     }
 
-    if (containsBlocked) {
+    if (blockedCount == 1 && this.allowBlockedEndDate_) {
```

Comment:
function is getting a bit long, would consider extracting out this logic and the above into a function that just checks for blocked days, since it seems to be orthogonal to the rest of the function.